linera-execution/src/revm.rs

```diff
@@ -244,6 +287,228 @@ pub enum KeyCategory {
     Storage,
 }
 
+// This is the precompile address that contains the LINERA specific
+// functionalities accessed from the EVM.
+fn precompile_address() -> Address {
+    address!("000000000000000000000000000000000000000b")
+}
+
+fn u8_slice_to_application_id(vec: &[u8]) -> ApplicationId {
+    let mut output = [0u64; 4];
+    for (i, chunk) in vec.chunks_exact(8).enumerate() {
+        output[i] = u64::from_be_bytes(chunk.try_into().unwrap());
+    }
+    let hash = output.into();
+    ApplicationId::new(hash)
+}
+
+fn address_to_user_application_id(address: Address) -> ApplicationId {
+    let address: Vec<u8> = address.to_vec();
+    let mut vec = vec![0_u8; 32];
+    vec[..20].copy_from_slice(&address[..20]);
+    u8_slice_to_application_id(&vec)
+}
+
+struct GeneralContractCall;
```

Comment:
maybe we could start thinking about splitting `revm.rs`

Reply:
So, I split off the `EvmExecutionError` and the `DatabaseRuntime` from it and created an `evm` directory.